### PR TITLE
upgraded to be compatible with nodebb 1.0.0

### PR DESCRIPTION
**Pull request by ckiss**

### CODE REVIEW

File: lib/auth.js [hunk 1]
```diff
@@ -4,9 +4,18 @@
 var db = module.parent.parent.require('./database'),
 	utils = require.main.require('./public/src/utils'),
 	async = require.main.require('async'),
+	url = require.main.require('url'),
 
 	Auth = {};
 
+Auth.mapExternalId = function(req, res, cb) {
+	var qs = url.parse(req.url, true).query;
+  db.getObjectField(qs.app + 'Id:uid', req.params.uid, function(err, uid) {
```
Comment (dberesford) [inline lib/auth.js]: can replace callback function  with `cb`

File: plugin.json [hunk 2]
```diff
@@ -9,12 +9,13 @@
...
 		{ "hook": "action:middleware.authenticate", "method": "authenticate"},
-		{ "hook": "action:settings.set", "method": "reloadSettings" }
+		{ "hook": "action:settings.set", "method": "reloadSettings" },
+		{ "hook": "filter:user.updateProfile", "method": "EnableUserFieldsToWrite" }
 	],
 	"less": [
 		"./less/style.less"
 	],
-	"scripts": [
+	"acpScripts": [
```
Comment (dberesford) [inline plugin.json:18]: acpScripts?

File: routes/v1/topics.js [hunk 3]
```diff
@@ -74,6 +54,29 @@ module.exports = function(middleware) {
...
+				return false;
+			}
+
+			var payload = {
+				uid: req.user.uid,
+				pid: req.body.pid,
+				content: req.body.content,
+				options: {}
+			};
+			console.log(payload);
```
Comment (dberesford) [inline routes/v1/topics.js]: remove console.log

**Comment by dberesford** [comment]
Minor nits but LGTM in general, good work @ckiss

File: plugin.json (hunk 2 above)
Comment (ckiss) [inline plugin.json:18]: this seems to be a thing in nodebb... I found some other guys doing the same in other plugins

**Pull request merged**